### PR TITLE
feat: add support in s3

### PR DESCRIPTION
**Pull request by AnguIar**

### CODE REVIEW

**Comment by AnguIar** [comment]
- [ ] S3 download links expire after 1 week **at max**, how do we solve this

File: confd/production.tmpl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "kafka" : {
         "host_ip": {{ getv "/configuration/exporter/worker/kafka/host/ip" "[\"10.28.11.49:9092\"]" }},
-        "topic": "{{ getv "/configuration/exporter/worker/kafka/topic" "topic-test-101" }}",
+        "topic": "{{ getv "/configuration/exporter/worker/kafka/topic" "topic-try-123" }}",
```
Comment (asafMasa) [inline confd/production.tmpl]: revert and erase the topic

File: confd/production.tmpl [hunk 2]
```diff
@@ -26,6 +26,13 @@
         "multithread": {{ getv "/configuration/exporter/worker/gdal/multithread" "true" }},
         "thread_count": {{ getv "/configuration/exporter/worker/gdal/thread/count" "-1" }}
     },
+    "s3": {
+        "endpoint_url": "{{ getv "/configuration/exporter/worker/s3/endpoint/url" "http://localhost:9000" }}",
+        "access_key_id": "{{ getv "/configuration/exporter/worker/s3/access/key/id" "minioadmin" }}",
+        "secret_access_key": "{{ getv "/configuration/exporter/worker/s3/secret/access/key" "minioadmin" }}",
+        "bucket": "{{ getv "/configuration/exporter/worker/s3/bucket" "mapping-images" }}"
+    },
+    "storage_provider": "{{ getv "/configuration/exporter/worker/storage/provider" "fs" }}",
```
Comment (asafMasa) [inline confd/production.tmpl:35]: insert "external_physical_path" and "internal_outputs_path"  under fs {} config object

Comment (asafMasa) [reply confd/production.tmpl:35]: rename "input_output" to something meaningful

File: run.sh [hunk 3]
```diff
@@ -1,2 +1,12 @@
 #!/bin/bash
-docker run -d --name exporter-worker -v /home/ubuntu/outputs:/app/outputs exporter-worker:latest
+#docker run -d --name exporter-worker -v /home/ubuntu/outputs:/app/outputs exporter-worker:latest
+docker run -d \
+-e CONFIGURATION_EXPORTER_WORKER_EXPORTSTORAGE_URL=http://192.168.49.1:8080 \
```
Comment (asafMasa) [inline run.sh]: revert

File: confd/production.tmpl (hunk 2 above)
Comment (asafMasa) [inline confd/production.tmpl:35]: add "storage_provider" usage to readme file

File: src/exportImage.py [hunk 4]
```diff
@@ -59,10 +61,17 @@ def export(self, bbox, filename, url, taskid, directoryName, max_zoom):
                 result = self.create_geopackage(
                     bbox, filename, url, taskid, full_path, resolution)
 
-                if result is not None:
+                if result:
                     self.create_index(filename, full_path)
+                    if (self.__config["storage_provider"] == StorageProvider.S3.value):
+                        self.upload_to_s3(filename, directoryName,
+                                          output_format, full_path)
+
```
Comment (asafMasa) [inline src/exportImage.py]: after upload do:
1. get full size // file_size = path.getsize(fullPath)
2.  self.delete_local_directory(directoryName)
3. send to save_update the full_size as an optional parametrer

**Review by asafMasa — CHANGES_REQUESTED** [review]
🥇 good job

File: README.md [hunk 5]
```diff
@@ -52,7 +52,7 @@ this will run container with your kafka topic declared topic value , in this cas
 
 **_Important:_**
 
-when running in production must provide config value: "INPUT_OUTPUT_EXTERNAL_PHYSICAL_PATH" (see below on configurations section) as full path, to declare the creation of the packages location,
+when running in production must provide config value: "FS_EXTERNAL_PHYSICAL_PATH" (see below on configurations section) as full path, to declare the creation of the packages location,
```
Comment (asafMasa) [inline README.md]: must? what if we choose s3?

**Review by asafMasa — CHANGES_REQUESTED** [review]
image
why?

**Comment by AnguIar** [comment]
> image
> why?

It says empty file but if you look into it (`view file`) you see that it's all good.

File: confd/production.tmpl (hunk 1 above)
Comment (asafMasa) [inline confd/production.tmpl]: revert

**Review by asafMasa — APPROVED** [review]

**Pull request merged**